pkg/vault/contracts/authorizer/TimelockAuthorizer.sol

```diff
@@ -850,8 +871,20 @@ contract TimelockAuthorizer is IAuthorizer, IAuthentication, ReentrancyGuard {
     ) external {
         require(isRoot(msg.sender), "SENDER_IS_NOT_ROOT");
 
-        bytes32 revokePermissionsActionId = getRevokePermissionActionId(actionId);
-        _revokePermission(revokePermissionsActionId, account, where);
+        require(isRevoker(actionId, account, where), "ACCOUNT_IS_NOT_REVOKER");
+
+        require(!isRoot(account), "CANNOT_REMOVE_ROOT_REVOKER");
+
+        // On top of requiring that the account is currently a revoker, we prevent attempts to remove permission over a
+        // single contract from global revokers. As mentioned in `addRevoker`, it is possible for an account to have
+        // both global and specific permissions over a given contract: in this case, the global permission must be
+        // removed before the specific ones can be addressed.
+        if (_isRevoker[actionId][account][EVERYWHERE]) {
+            require(where == EVERYWHERE, "REVOKER_IS_GLOBAL");
```

Comment:
Maybe "ACCOUNT_IS_GLOBAL_REVOKER" ?

Reply:
I've picked this for consistency with `require(where == EVERYWHERE, "GRANTER_IS_GLOBAL");`. I think `ACCOUNT_IS_GLOBAL_REVOKER` is better but I would change both of them in a separate PR.